sklearn/compose/tests/test_target.py

```diff
@@ -265,3 +267,26 @@ def test_transform_target_regressor_ensure_y_array():
     tt.predict(X.tolist())
     assert_raises(AssertionError, tt.fit, X, y.tolist())
     assert_raises(AssertionError, tt.predict, X)
+
+
+def test_transform_target_regressor_allow_nan():
+    # check if the TransformedTargetRegressor allows missing value in the target array
+
+    X, y = datasets.load_linnerud(return_X_y=True)
+
+    # put some NaN in y
+    y[5, 1] = np.NaN
+
+    class DummyInvertibleImputer(SimpleImputer):
+        """ add a dummy inverse transform to simple impute"""
+        def inverse_transform(self, X):
+            return X
+
+    estimator = TransformedTargetRegressor(
+        regressor=LinearRegression(),
+        transformer=DummyInvertibleImputer(),
+        check_inverse=False
+    )
+
+    estimator.fit(X, y)
+    estimator.predict(X)
```

Comment:
You should check that your output contains NaN where it should.

Reply:
`estimator.predict(X)` is actually unnecessary. The test only to assert that `estimator.fit(X, y)` doesn't raise. In this case, predict is expected to return no NaN. should I make it explicit or remove `predict`?